### PR TITLE
[WIP]Build and scan latest etcd commit

### PR DESCRIPTION
**Pull request by ArkaSaha30**
This PR will add trivy nightly scan for release-3.5 to build from latest commit and scan for vulnerabilities.

Fixes: https://github.com/etcd-io/etcd/issues/14991

Signed-off-by: ArkaSaha30 [arkasaha30@gmail.com](mailto:arkasaha30@gmail.com)

### CODE REVIEW

**Comment by ArkaSaha30** [comment]
/assign @ahrtr 

I am not sure if this is exactly what we want to do, but I wanted to start the conversation. I tested this with my personal fork and the distroless images did not have any CVEs. 
Am I missing anything?

**Comment by ahrtr** [comment]
Thanks @ArkaSaha30 for the PR.

1. I think we need to do it for the main branch firstly. We can add one more step in the existing release.yml, and scan the image directly (no need to build the image again). Similar to what https://github.com/etcd-io/etcd/pull/15350 is doing. Probably this PR should be on top of it.
2. We need to do similar thing for release-3.5 and release-3.4. But the release workflow/script in 3.5 and 3.4 are always running against release-3.5 and release-3.4 respectively instead of the dev branch. So probably we need to backport https://github.com/etcd-io/etcd/pull/14860 to 3.5 and 3.4 firstly.

**Comment by ArkaSaha30** [comment]
Thank you @ahrtr for the review. 

1. Before I open a separate PR to `etcd:main`, I wanted to confirm if step:`trivy-scan` would come right after `test-image` and before `publish-release`(once `release.sh` is modularised)
Ref: https://github.com/etcd-io/etcd/pull/15350#discussion_r1122624175

2. For the backport of #14860 can I create a separate PR right-away with proper description and references?

**Comment by ahrtr** [comment]
> 1. Before I open a separate PR to `etcd:main`, I wanted to confirm if step:`trivy-scan` would come right after `test-image` and before `publish-release`(once `release.sh` is modularised)

Yes.


> For the backport of https://github.com/etcd-io/etcd/pull/14860 can I create a separate PR right-away with proper description and references?

Sounds good to me.

File: .github/workflows/trivy-nightly-scan.yaml [hunk 1]
```diff
@@ -1,37 +1,51 @@
...
         with:
           ref: release-3.5
-      - name: Run Trivy vulnerability scanner
-        uses: aquasecurity/trivy-action@9ab158e8597f3b310480b9a69402b419bc03dbd5 # master
+      - uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
         with:
-          image-ref: 'gcr.io/etcd-development/etcd:${{ matrix.versions }}'
+          go-version: "1.19.6"
+      - run: |
+            ./build.sh
```
Comment (pchan) [inline .github/workflows/trivy-nightly-scan.yaml:31]: We can remove this entire step/ There is no need to build the binaries or docker images, we can re-use the one created as part of release-workflow (once you have ported this to etcd:main). See this [code](https://github.com/etcd-io/etcd/blob/8b93e0e46bb0a698e425d8a4b0ba82f7510fd0c8/scripts/test_images.sh#L39-L56) for reference.

**Comment by jmhbnz** [comment]
Closing during triage for longstanding open pr's.  As mentioned above, this feature needs to be implemented in `main` before any backport to `release-3.5` could be considered.

**Pull request closed without merging**